pkg/router/router_test.go

```diff
@@ -190,17 +200,144 @@ func TestConfigTemplateExecution(t *testing.T) {
-				tlsTermination: routev1.TLSTerminationEdge,
+				configSnippet: "acl whitelist src -f " + filepath.Join(h.dirs["whitelist"], h.namespace+":a.txt"),
+			},
+		},
+		"Simple HSTS header": {
+			mustCreateWithConfig{
+				mustCreate: mustCreate{
+					name: "b",
+					host: "bexample.com",
+					path: "",
+					time: start,
+					annotations: map[string]string{
+						"haproxy.router.openshift.io/hsts_header": "max-age=99999;includeSubDomains;preload",
+					},
+					tlsTermination: routev1.TLSTerminationEdge,
+				},
+				configSnippet: `http-response set-header Strict-Transport-Security 'max-age=99999;includeSubDomains;preload'`,
+			},
+		},
+		"Simple HSTS header 2": {
+			mustCreateWithConfig{
+				mustCreate: mustCreate{
+					name: "b2",
+					host: "b2example.com",
+					path: "",
+					time: start,
+					annotations: map[string]string{
+						"haproxy.router.openshift.io/hsts_header": "max-age=99999;includeSubDomains",
+					},
+					tlsTermination: routev1.TLSTerminationEdge,
+				},
+				configSnippet: `http-response set-header Strict-Transport-Security 'max-age=99999;includeSubDomains'`,
+			},
+		},
+		"Case insensitive, with white spaces HSTS header": {
+			mustCreateWithConfig{
+				mustCreate: mustCreate{
+					name: "c",
+					host: "cexample.com",
+					path: "",
+					time: start,
+					annotations: map[string]string{
+						"haproxy.router.openshift.io/hsts_header": "max-age=99999 ;  includesubdomains;  PREload",
+					},
+					tlsTermination: routev1.TLSTerminationEdge,
+				},
+				configSnippet: `http-response set-header Strict-Transport-Security 'max-age=99999 ;  includesubdomains;  PREload'`,
+			},
+		},
+		"Quotes in HSTS header": {
+			mustCreateWithConfig{
+				mustCreate: mustCreate{
+					name: "d",
+					host: "dexample.com",
+					path: "",
+					time: start,
+					annotations: map[string]string{
+						"haproxy.router.openshift.io/hsts_header": `max-age="99999"`,
+					},
+					tlsTermination: routev1.TLSTerminationEdge,
+				},
+				configSnippet: `http-response set-header Strict-Transport-Security 'max-age="99999"'`,
+			},
+		},
+		"Equal sign with LWS in HSTS header": {
+			mustCreateWithConfig{
+				mustCreate: mustCreate{
+					name: "f",
+					host: "fexample.com",
+					path: "",
+					time: start,
+					annotations: map[string]string{
+						"haproxy.router.openshift.io/hsts_header": `max-age  =  "99999"`,
+					},
+					tlsTermination: routev1.TLSTerminationEdge,
+				},
+				configSnippet: `http-response set-header Strict-Transport-Security 'max-age  =  "99999"'`,
+			},
+		},
+		"Required directive missing": {
+			mustCreateWithConfig{
+				mustCreate: mustCreate{
+					name: "g",
+					host: "gexample.com",
+					path: "",
+					time: start,
+					annotations: map[string]string{
+						"haproxy.router.openshift.io/hsts_header": "min-age=99999",
+					},
+					tlsTermination: routev1.TLSTerminationEdge,
+				},
+				configSnippet: `http-response set-header Strict-Transport-Security 'min-age=99999'`,
```

Comment:
Can we check for just the substring "http-response set-header Strict-Transport-Security" to make sure there are no false negatives?
```suggestion
				configSnippet: `http-response set-header Strict-Transport-Security`,
```

Reply:
Oh, the state isn't cleared between test cases, is it?  So with my suggestion, this test case would always fail.